tests/e2e_all/scripts/common/run_single_test.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# N.B. this file must not depend on any external environment variables that are set in various places by the end2end
+# test suite. When running under GNU parallels, the environment is empty, thus every single variable in this file must
+# either be explicitly set in the parallels command with `--env` or passed to this script as part of the argument list
```

Comment:
This script was previously the inner body of the run_tests loop, separated out to a standalone script with minimal dependencies so it can be parallelized